saos-search/src/main/java/pl/edu/icm/saos/search/search/service/FieldFacetToQueryApplier.java

```diff
@@ -15,19 +18,33 @@
 @Service
 public class FieldFacetToQueryApplier {
 
+    private FieldValuePrefixAdder fieldValuePrefixAdder;
+    
     
     //------------------------ LOGIC --------------------------
     
     /**
      * Applies field facet request to {@link SolrQuery}
      */
-    public void applyFieldFacet(SolrQuery query, String fieldName) {
+    public void applyFieldFacet(SolrQuery query, String fieldName, String fieldValuePrefix) {
         
         Preconditions.checkNotNull(query);
         Preconditions.checkNotNull(fieldName);
         Preconditions.checkArgument(StringUtils.isNotBlank(fieldName));
         
         query.setFacet(true);
+        query.add(FacetParams.FACET_SORT, FacetParams.FACET_SORT_INDEX);
         query.add(FacetParams.FACET_FIELD, fieldName);
+        if (!StringUtils.isBlank(fieldValuePrefix)) {
+            query.add(FacetParams.FACET_PREFIX, fieldValuePrefixAdder.prefixWithSeparator(fieldValuePrefix));
+        }
```

Comment:
Please consider adding params FacetParams.FACET_SORT, FacetParams.FACET_PREFIX for single field. That is `f.<fieldname>.facet.sort` and `f.<fieldname>.facet.prefix`.
In form like this second call of applyFieldFacet for different field would not override params from first call.